packages/jsii-dotnet-analyzers/src/Amazon.JSII.Analyzers/JsiiOptionalAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Amazon.JSII.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class JsiiOptionalAnalyzer : DiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "CDK001";
+        private const string Title = "A required property is missing or null";
+        private const string MessageFormat = "The property is required and can not be null";
+        private const string MessageFormatWithPropertyName = "The property {0} is required and can not be null";
+        private const string Description = "The property is required and can not be null";
+        private const string DescriptionWitPropertyName = "The property {0} is required and can not be null";
```

Comment:
For all of those, I'd have written as `cannot` instead of `can not`. Maybe just me?

Reply:
Both are acceptable, but I agree that the first one is more common